src/blueapi/core/context.py

```diff
@@ -97,16 +100,22 @@ def plan_2(...) -> MsgGenerator:
 
         for obj in load_module_all(module):
             if is_bluesky_plan_generator(obj):
-                self.plan(obj)
+                self.register_plan(obj)
 
     def with_device_module(self, module: ModuleType) -> None:
         self.with_dodal_module(module)
 
+    def get_lazy_devices(self) -> DevicesDict:
+        return {k: v for k, v in self.devices.items() if v.lazy}
+
     def with_dodal_module(self, module: ModuleType, **kwargs) -> None:
         devices, exceptions = make_all_devices(module, **kwargs)
 
-        for device in devices.values():
-            self.device(device)
+        # for non-lazy devices, we instantiate them
+        early_devices = devices.items().filter(lambda x: not x.lazy)
```

Comment:
Nit: For consistency I suggest we use the blanket term "eager" in place of "early" and "non-lazy"

Reply:
done